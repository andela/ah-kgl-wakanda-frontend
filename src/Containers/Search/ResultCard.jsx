import React from 'react';
import PropTypes from 'prop-types';
import { Card, Image } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';
import * as paths from '../../paths';
import Button from '../../Components/Common/Button/Button';
import blankProfilePic from '../../assets/img/blank_profile_pic.png';
import blackArticleImg from '../../assets/images/undraw_article.png';
import './ResultCard.scss';

/**
 * display search results into cards
 * @param {object} props
 * @returns {object} jsx
 */
const ResultCard = ({ avatar, username, date, articleImg, title, description, slug }) => (
  <Card className="article-card">
    <Card.Header>
      <div className="d-flex flex-row">
        <Image className="avatar" src={avatar} roundedCircle />
        <div className="d-flex flex-column ml-2">
          <span className="username">{username}</span>
          <span className="date">{date}</span>
        </div>
      </div>
    </Card.Header>
    <Card.Body className="d-flex flex-row">
      <div>
        <Card.Img variant="top" className="article-img" src={articleImg} />
      </div>
      <div className="article-info">
        <Card.Title className="title">{title}</Card.Title>
        <Card.Subtitle className="mb-2 subtitle">{description}</Card.Subtitle>
        <a href={`${paths.ARTICLES_PATH}/${slug}`}>
          <Button size={12} text="View more..." />
        </a>
      </div>
    </Card.Body>
  </Card>
);

ResultCard.propTypes = {
  avatar: PropTypes.string,
  articleImg: PropTypes.string,
  slug: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

ResultCard.defaultProps = {
  avatar: blankProfilePic,
  articleImg: blackArticleImg,
};

export default ResultCard;
