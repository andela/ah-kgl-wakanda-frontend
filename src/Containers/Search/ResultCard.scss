@import '../../assets//sass/global.scss';
.results {
  margin-top: 120px;
  margin-left: 120px;
  margin-right: 120px;
  .result-column {
    .article-card {
      margin-bottom: 30px;
      border: 0;

      .card-header {
        padding: 0.75rem 0;
      }

      .card-body {
        padding-left: 0;
        margin-top: 12px;

        .card-img-top {
          border-top-left-radius: 0;
          border-top-right-radius: 0;
        }
      }

      .article-info {
        margin-left: 20px;
        margin-top: 15px;
        .title {
          padding-bottom: 18px;
          font-size: 1.8em;
          font-weight: 700;
        }

        .subtitle {
          font-size: 1.2em;
        }

        .ah-button {
          margin: 15px 0;
        }
      }

      .card-header {
        background: #fff;
      }
      .avatar {
        width: 60px;
        height: 60px;
        padding: 3px;
        border-radius: 30px;
        border: 2px solid $color-primary;
      }

      .username {
        font-size: 18px;
        text-transform: capitalize;
      }

      .date {
        font-size: 13px;
        color: #666;
      }

      .article-img {
        background-size: cover;
        width: 280px;
        height: 200px;
      }
    }
  }
}

@media #{$extra-small-device} {
  .results {
    margin-left: 0;
    margin-right: 0;

    .result-column {
      .article-card {
        .article-img {
          width: 120px;
          height: 90px;
        }
      }
    }
  }
}

@media #{$small-device} {
  .results {
    margin-left: 0;
    margin-right: 0;

    .article-img {
      width: 50px;
      height: 20px;
    }
  }
}

@media #{$medium-device} {
  .results {
    margin-left: 0;
    margin-right: 0;
  }
}

@media #{$large-device} {
}

@media #{$extra-large-device} {
}
