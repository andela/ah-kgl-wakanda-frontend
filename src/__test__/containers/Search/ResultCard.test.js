import React from 'react';
import { shallow } from 'enzyme';
import ResultCard from '../../../Containers/Search/ResultCard';

describe('ResultCard component', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      slug: 'slug',
      username: 'test',
      date: '12 jun 2019',
      title: 'articles',
      description: 'lorem ipsum',
    };
    wrapper = shallow(<ResultCard {...props} />);
  });

  it('should render ResultCard component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
