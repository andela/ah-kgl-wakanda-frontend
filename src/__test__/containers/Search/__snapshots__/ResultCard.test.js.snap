// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard component should render ResultCard component 1`] = `ShallowWrapper {}`;
